commit 733ddf5edff5f18609d0add0b76da764111f7f1e
Author: Dmitry Grinberg <dmitrygr@gmail.com>
Date:   Fri Dec 30 00:39:11 2022 -0600

    bugfix: crash in weird cases
    
    If you have an image database that has some images but not others, the
    app will crash on OS6.0 and some OS5.x devices if you first view a poke
    without an image, and then one with

diff --git a/Src/PkmnMain.c b/Src/PkmnMain.c
index 666b01b..c96c954 100644
--- a/Src/PkmnMain.c
+++ b/Src/PkmnMain.c
@@ -37,7 +37,7 @@ void DrawPkmnSprite(UInt16 selectedPkmnId)
 	// Check if the PNG for the current pkmn
 	// is already decoded in memory
 	error = FtrGet(appFileCreator, 0, (UInt32*)&ds);
-	if (error == errNone)
+	if (error == errNone && ds)
 	{
 		// If it is, draw it and return
 		imgDrawRedraw(ds, POKE_IMAGE_AT_X, POKE_IMAGE_AT_Y);
